linux/runtime.go

```diff
@@ -383,7 +383,17 @@ func (r *Runtime) loadTasks(ctx context.Context, ns string) ([]*Task, error) {
 		)
 		ctx = namespaces.WithNamespace(ctx, ns)
 		pid, _ := runc.ReadPidFile(filepath.Join(bundle.path, proc.InitPidFile))
-		s, err := bundle.NewShimClient(ctx, ns, ShimConnect(), nil)
+		s, err := bundle.NewShimClient(ctx, ns, ShimConnect(func() {
+			log.G(ctx).WithError(err).WithFields(logrus.Fields{
+				"id":        id,
+				"namespace": ns,
+			}).Error("connecting to shim")
+			err := r.cleanupAfterDeadShim(ctx, bundle, ns, id, pid)
```

Comment:
missing a `task.Delete()` here.

If it's not done, re-running the same container (with the same name) will fail until containerd is restarted (at which point it'll delete the task when the plugin loads)

Reply:
but we don't have a task at this point.  Shouldn't r.terminate do the same thing?

Reply:
That why I moved the task adding in my orig PR.

The `task.Delete()` call is after the call to `cleanupAfterDeadShim` if it doesn't error out. I think it is an issue now, because task with a dead shim can no longer be listed as when a list of task is requested we now try to get its state from the shim. The original behavior left it to the user to invoke the Delete call, but that's no longer possible.

It may be better to automatically delete the task in r.terminate. But the proper event would need to be generated too (realized that wasn't the case in my orig PR either).